tools/dmake/dmake.cpp

```diff
@@ -168,9 +169,9 @@ static std::string getCppFiles(std::vector<std::string> &files, const std::strin
         return err;
 
     // add *.cpp files to the "files" vector..
-    for (const std::pair<const std::string&, size_t> file : filelist) {
```

Comment:
The previous type was wrong and there was also an unnecessary copy which was not detected by clang-tidy.